drivers/display/ft8xx/ft8xx.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020 Hubert Miś
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/display/ft8xx/ft8xx.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <kernel.h>
+#include <logging/log.h>
+
+#include <drivers/display/ft8xx/ft8xx_copro.h>
+#include <drivers/display/ft8xx/ft8xx_dl.h>
+
+#include "ft8xx_common.h"
+#include "ft8xx_drv.h"
+#include "ft8xx_host_commands.h"
+#include "ft8xx_memory.h"
+
+#define DT_DRV_COMPAT ftdi_ft800
+
+#define LOG_MODULE_NAME ft8xx
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define PCLK      DT_INST_PROP(0, pclk)
+#define PCLK_POL  DT_INST_PROP(0, pclk_pol)
+#define CSPREAD   DT_INST_PROP(0, cspread)
+#define SWIZZLE   DT_INST_PROP(0, swizzle)
+#define VSIZE     DT_INST_PROP(0, vsize)
+#define VOFFSET   DT_INST_PROP(0, voffset)
+#define VCYCLE    DT_INST_PROP(0, vcycle)
+#define VSYNC0    DT_INST_PROP(0, vsync0)
+#define VSYNC1    DT_INST_PROP(0, vsync1)
+#define HSIZE     DT_INST_PROP(0, hsize)
+#define HOFFSET   DT_INST_PROP(0, hoffset)
+#define HCYCLE    DT_INST_PROP(0, hcycle)
+#define HSYNC0    DT_INST_PROP(0, hsync0)
+#define HSYNC1    DT_INST_PROP(0, hsync1)
+
+#define DLSWAP_FRAME 0x02
+
+#define FT8XX_EXPECTED_ID 0x7C
+
+static int_callback irq_callback;
+
+static void command(uint8_t cmd)
+{
+	(void)ft8xx_drv_command(cmd);
+}
+
+static void busy_wait(void)
+{
+	k_sleep(K_MSEC(20));
+}
+
+static bool verify_chip(void)
+{
+	uint32_t id = rd32(REG_ID);
+
+	return (id & 0xff) == FT8XX_EXPECTED_ID;
+}
+
+static int ft8xx_init(const struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+	int ret;
+
+	ret = ft8xx_drv_init();
+	if (ret < 0) {
+		LOG_ERR("FT8xx driver initialization failed with %d", ret);
+		return ret;
+	}
+
+	/* Reset display controller */
+	command(CORERST);
+	command(ACTIVE);
+	busy_wait();
+	command(CLKEXT);
+	command(CLK48M);
+	busy_wait();
+
+	command(CORERST);
+	command(ACTIVE);
+	busy_wait();
+	command(CLKEXT);
+	command(CLK48M);
+	busy_wait();
+
+	while (!verify_chip()) {
+		/* Intentionally empty */
```

Comment:
can you explain why is this polling needed?

Reply:
It helps debugging hardware issues. If FT800 chip is incorrectly defined in device tree or incorrectly connected, the initialization procedure stops in this loop. I think it could be replaced with an error log